### PR TITLE
OCPBUGSM-20425 - failed day2 host should be able to be reset and to r…

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
…e-register

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -1634,6 +1634,52 @@ paths:
           description: Error.
           schema:
             $ref: '#/definitions/error'
+  /clusters/{cluster_id}/hosts/{host_id}/actions/reset:
```
Comment (filanov) [inline swagger.yaml:1637]: Isn't it part of reset cluster?
@avishayt do you think that we need to add another endpoint to reset day 2 hosts?

Comment (yevgeny-shnaidman) [reply swagger.yaml:1637]: @filanov , in day2 cluster is just a wrap around for hosts. each hosts is independent of another. there are not action on a cluster, only on individual host. One host maybe just installing, another discovering, and another just failed to install, and the cluster stays the same

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -2942,6 +2942,44 @@ func (b *bareMetalInventory) ResetCluster(ctx context.Context, params installer.
...
+		}
+		log.WithError(err).Errorf("failed to get host %s", params.HostID)
+		msg := fmt.Sprintf("Failed to reset host %s: error fetching host from DB", params.HostID.String())
+		b.eventsHandler.AddEvent(ctx, params.ClusterID, &params.HostID, models.EventSeverityError, msg, time.Now())
+		return common.NewApiError(http.StatusInternalServerError, err)
+	}
+
+	if !host.IsDay2Host(&h) {
+		log.Errorf("ResetHost for host %s is forbiddent: not a Day2 hosts", params.HostID.String())
+		return common.NewApiError(http.StatusForbidden, fmt.Errorf("hosts is not day2"))
```
Comment (avishayt) [inline internal/bminventory/inventory.go:2963]:
```suggestion
		return common.NewApiError(http.StatusConflict,, fmt.Errorf("Method only allowed when adding hosts to an existing cluster"))
```

File: internal/bminventory/inventory_test.go [hunk 3]
```diff
@@ -3945,6 +3945,78 @@ var _ = Describe("Register AddHostsCluster test", func() {
...
+
+	It("Reset day2 host, host is not day2 host", func() {
+		params := installer.ResetHostParams{
+			HTTPRequest: request,
+			ClusterID:   clusterID,
+			HostID:      hostID,
+		}
+		addHost(hostID, models.HostRoleWorker, models.HostStatusKnown, models.HostKindHost, clusterID, getInventoryStr("hostname0", "bootMode", "1.2.3.4/24", "10.11.50.90/16"), db)
+		res := bm.ResetHost(ctx, params)
+		verifyApiError(res, http.StatusForbidden)
```
Comment (avishayt) [inline internal/bminventory/inventory_test.go:4016]:
```suggestion
		verifyApiError(res, http.StatusConflict)
```

File: swagger.yaml [hunk 4]
```diff
@@ -1634,6 +1634,52 @@ paths:
...
+          schema:
+            $ref: '#/definitions/infra_error'
+        404:
+          description: Error.
+          schema:
+            $ref: '#/definitions/error'
+        405:
+          description: Method Not Allowed.
+          schema:
+            $ref: '#/definitions/error'
```
Comment (avishayt) [inline swagger.yaml:1674]: 405 not needed

**Review by avishayt — CHANGES_REQUESTED** [review]

File: internal/bminventory/inventory.go (hunk 2 above)
Comment (yevgeny-shnaidman) [reply internal/bminventory/inventory.go]: fixed

File: internal/bminventory/inventory_test.go (hunk 3 above)
Comment (yevgeny-shnaidman) [reply internal/bminventory/inventory_test.go]: fixed

File: swagger.yaml (hunk 4 above)
Comment (yevgeny-shnaidman) [reply swagger.yaml]: fixed

**Review by avishayt — APPROVED** [review]

**Pull request merged**